rpc/documentation/api.md

```diff
@@ -729,6 +731,102 @@ transaction paying to already known addresses or scripts.
+
+- `uint32 account`: Account number containing the keys controlling the output
+  set to query.
+
+- `repeated Output outputs`: A list of outputs to include with the transaction.
+  These outputs take an address string rather than a script for ease of use.
+  
+   **Nested message:** `Output`
+  
+    - `string address`: The address for this output to pay.
+  
+    - `int64 amount`: The amount to pay.
+
+- `int32 required_confirmations`: The minimum number of block confirmations
+  needed to consider including an output in the return set.  This may not be
+  negative.
+
+- `uint32 sat_per_kb_fee`: The fee to pay in satoshis per kilobyte.
+
+**Response:** `CreateTransactionResponse`
+
+- `bytes serialized_transaction`: The serialized transaction with the inputs and
+  outputs added.
+
+- `repeated int64 input_values`: The value (in satoshis) of each input. This is
+  needed to sign the transaction using the bitcoin cash signing algorithm.
+  
+- `int64 fee`: The fee that ended up being set when the transaction was created.
+
+**Expected errors:**
+
+- `InvalidArgument`: The target amount is negative.
+
+- `InvalidArgument`: The required confirmations is negative.
+
+- `Aborted`: The wallet database is closed.
+
+- `NotFound`: The account does not exist.
+
+**Stability:** Unstable
+
+___
+
+#### `SweepAccount`
+
+The `SweepAccount` method provides a function to sweep the full amount of funds
+in the account to a new address. The reason this function is needed is because
+`CreateTransaction` will attempt to add the fee to the requested amount, posssibly
+pushing it over the total balance in the wallet and resulting in a insufficient funds
+error. This requires the user to play a guessing game where they try to guess the
+exact fee the wallet will use and subtract that from their balance when creating a 
+spend. It's much easier to just have a sweep account function which grabs all the 
+inputs in the wallet, totals them up, and subtracts the fee from the total. So
+unlikely `CreateTransaction` the fee here is subtracted from the total amount
+rather than added.
+
+**Request:** `SweepAccount`
+
+- `uint32 account`: Account number containing the keys controlling the output
+  set to query.
+
+- `string sweep_to_address`: The address to sweep all the funds into.
+
+- `uint32 sat_per_kb_fee`: The fee to pay in satoshis per kilobyte.
+
+**Response:** `SweepAccountResponse`
+
+- `bytes serialized_transaction`: The serialized transaction with the inputs and
+  outputs added.
+
+- `repeated int64 input_values`: The value (in satoshis) of each input. This is
+  needed to sign the transaction using the bitcoin cash signing algorithm.
+  
+- `int64 total_amount`: The total amount that ended up being swept to the address.
+  
+- `int64 fee`: The fee that ended up being set when the transaction was created.
+
+**Expected errors:**
+
+- `Aborted`: The wallet database is closed.
+
+- `NotFound`: The account does not exist.
+
+**Stability:** Unstable
```

Comment:
Curious, what is our litmus test on when to change the Stability rating?

Reply:
Yeah I noticed that. They labeled the entire gRPC API as unstable. I think unstable in their parlance means it may change in the future.